Update and split troubleshooting page (#512)
foundriesio/docs: source/user-guide/troubleshooting/troubleshooting.rst

       "

       do_install:append() {
           install -Dm 0644 ${WORKDIR}/fioconfig.conf ${D}${sysconfdir}/default/fioconfig
       }

9. Create another directory under the one we just created so we can supply the source file (``fioconfig.conf``) for the recipe above:

   .. prompt:: bash host:~$

       cd meta-subscriber-overrides
       mkdir -p recipes-support/fioconfig/fioconfig

10. Create the ``fioconfig.conf`` file under this new directory including:

   .. code-block::

       DAEMON_INTERVAL=<time-sec>

.. note::
    Make sure to replace ``<time-sec>`` with the expected poll interval in seconds.

Commit and trigger a new build to include these new changes and have a new polling interval.

Re-Register a Device
^^^^^^^^^^^^^^^^^^^^

During development, you may need to re-register the same device.
Follow these steps to do so:

1. Delete the device from the UI ``Devices`` tab or with:

   .. prompt:: bash host:~$

       fioctl device delete <device-name>

2. Stop ``aktualizr-lite`` and ``fioconfig`` on the device:

   .. prompt:: bash device:~#

       systemctl stop aktualizr-lite
       systemctl stop fioconfig.path
       systemctl stop fioconfig.service

> 3. Delete ``sql.db`` on the device:

Reviewer: you need to replace sql.db with sota.toml since that is now the indication that the device is registered.

Author: @Tim-Anderson do you mean on line 288 and 292 the mentions of ``sql.db`` should read ``sota.toml``?

All of your Targets contained in :term:`targets.json` can grow large over time::

  Signing local TUF targets
  == 2020-11-24 23:12:53 Running: garage-sign targets sign --repo /root/tmp.dNLAIH
  --key-name targets
  |  signed targets.json to /root/tmp.dNLAIH/roles/targets.json
  |--
  Publishing local TUF targets to the remote TUF repository
  == 2020-11-24 23:12:55 Running: garage-sign targets push --repo /root/tmp.dNLAIH
  |  An error occurred
  |  com.advancedtelematic.libtuf.http.SHttpjServiceClient$HttpjClientError:
  ReposerverHttpClient|PUT|http/413|https://api.foundries.io/ota/repo/magicman//api/v1/user_repo/targets|<html>
  |  <head><title>413 Request Entity Too Large</title></head>
  |  <body>
  |  <center><h1>413 Request Entity Too Large</h1></center>
  |  <hr><center>nginx/1.19.3</center>
  |  </body>
  |  </html>

Solution
""""""""

Over time, the manual pruning (deletion) of Targets is maintenance you should consider.

Pruning can be done using Fioctl™.
This removes outdated Targets from your Factory's :term:`targets.json`, allowing the production of new Targets.

.. warning::

   Ensure there are no important devices running on a Target that is about to be pruned.

>    If you are intending on pruning ``master``, be cautious and mindful of what you are doing.

Reviewer 2: ```suggestion
   If you are intending on pruning production tags, be cautious and mindful of what you are doing.
```

       }

9. Create another directory under the one we just created so we can supply the source file (``fioconfig.conf``) for the recipe above:

   .. prompt:: bash host:~$

       cd meta-subscriber-overrides
       mkdir -p recipes-support/fioconfig/fioconfig

10. Create the ``fioconfig.conf`` file under this new directory including:

   .. code-block::

       DAEMON_INTERVAL=<time-sec>

.. note::
    Make sure to replace ``<time-sec>`` with the expected poll interval in seconds.

Commit and trigger a new build to include these new changes and have a new polling interval.

Re-Register a Device
^^^^^^^^^^^^^^^^^^^^

During development, you may need to re-register the same device.
Follow these steps to do so:

1. Delete the device from the UI ``Devices`` tab or with:

   .. prompt:: bash host:~$

       fioctl device delete <device-name>

2. Stop ``aktualizr-lite`` and ``fioconfig`` on the device:

   .. prompt:: bash device:~#

       systemctl stop aktualizr-lite
       systemctl stop fioconfig.path
       systemctl stop fioconfig.service

3. Delete both ``sql.db`` and ``client.pem`` on the device:

   .. prompt:: bash device:~#

       rm /var/sota/sql.db

>        rm /var/sota/client.pem

Author: @Tim-Anderson @mike-sul does this look correct?

Reviewer 2: @kprosise I have a ticket to fix this part, so if you want, you can leave as it is for now. We need to extend with some warnings there, it depends if your device is open/closed, etc.

Note only a handle to the private key in the Secure Element is provided and stored by the TA.

During that creation process the TA also keeps a copy of the key on its heap.

This means that a system that creates all of its keys during boot may run out of heap before running out of SE storage.

To avoid this issue, configure OP-TEE with a large enough ``CFG_PKCS11_TA_HEAP_SIZE``.
It should allow the client to fill the SE NVM before an out of memory condition is raised by the TA.
This will help avoid a secure world panic.

An experimental way to validate the thresholds is to loop on RSA or EC key creation until it fails.
If there is a panic or a PKCS#11 OOM fault, ``CFG_PKCS11_TA_HEAP_SIZE`` can then be increased as there is still room in the SE NVM to store more keys.

.. _ref-troubleshooting_network-connectivity:

Debugging Network Connectivity
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When debugging network connectivity and access issues, it can be helpful to use ``curl``.
However, LmP does not ship with the command.

Rather than including ``curl`` on the host device, a simple approach is to run it via a Alpine Linux® container.

> 

Reviewer 2: Need to double check formatting:
```suggestion
    docker run --rm -it alpine
    / # apk add curl
    / # curl
```

Updates To etc
^^^^^^^^^^^^^^

Files created or modified in ``/etc`` during runtime are not handled by OSTree during an OTA.

> For this reason, we suggest setting system-wide configs in ``/usr`` whenever possible so that these changes are covered by OTA updates.

Reviewer 2: ```suggestion
For this reason, we suggest setting system-wide configs in ``/usr`` rather than ``/etc`` whenever possible so that these changes are covered by OTA updates.
```

> We suggest managing files that live in ``/usr`` with a systemd service (:ref:`ref-troubleshooting_systemd-service`).

Reviewer 2: ```suggestion
We suggest managing files that live in ``/etc`` with a systemd service (:ref:`ref-troubleshooting_systemd-service`).
```

During that creation process the TA also keeps a copy of the key on its heap.

This means that a system that creates all of its keys during boot may run out of heap before running out of SE storage.

To avoid this issue, configure OP-TEE with a large enough ``CFG_PKCS11_TA_HEAP_SIZE``.
It should allow the client to fill the SE NVM before an out of memory condition is raised by the TA.
This will help avoid a secure world panic.

An experimental way to validate the thresholds is to loop on RSA or EC key creation until it fails.
If there is a panic or a PKCS#11 OOM fault, ``CFG_PKCS11_TA_HEAP_SIZE`` can then be increased as there is still room in the SE NVM to store more keys.

.. _ref-troubleshooting_network-connectivity:

Debugging Network Connectivity
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

When debugging network connectivity and access issues, it can be helpful to use ``curl``.
However, LmP does not ship with the command.

Rather than including ``curl`` on the host device, a simple approach is to run it via a Alpine Linux® container.
    docker run --rm -it alpine
    / # apk add curl
    / # curl

> Updates To etc

Author: ```suggestion

Updates To etc
```
